### PR TITLE
feat: change logger to zap.Logger

### PR DESCRIPTION
**Pull request by RainbowDashy**
Close BYT-749

### CODE REVIEW

File: go/base/log.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package base
+
+import (
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type Log struct {
```
Comment (dragonly) [inline go/base/log.go:10]: `Logger` may be more appropriate.

File: go/cmd/gh-ost/main.go [hunk 2]
```diff
@@ -136,6 +136,7 @@ func main() {
 	quiet := flag.Bool("quiet", false, "quiet")
 	verbose := flag.Bool("verbose", false, "verbose")
 	debug := flag.Bool("debug", false, "debug mode (very verbose)")
+	_ = flag.Bool("stack", false, "add stack trace upon error")
```
Comment (dragonly) [inline go/cmd/gh-ost/main.go:139]: Please add some detailed message saying it's a compatibility flag for gh-ost, and actually we will always add stack info at the error logging level.

**Review by dragonly — CHANGES_REQUESTED** [review]

**Comment by dragonly** [comment]
LG otherwise.

File: go/base/log.go (hunk 1 above)
Comment (RainbowDashy) [reply go/base/log.go]: Done

File: go/cmd/gh-ost/main.go (hunk 2 above)
Comment (RainbowDashy) [reply go/cmd/gh-ost/main.go:140]: Done

**Comment by RainbowDashy** [comment]
PTAL

**Review by dragonly — APPROVED** [review]

**Pull request merged**